Static/AnalysisArchitecture.hs

```diff
@@ -927,7 +953,8 @@ lambda expressions, like you do in the following -}
    do
      -- beh will be ignored for now
      (_rsig@(BranchRefSig _ (usig, _)), dg', asp') <-
-           anaUnitSpec lgraph libEnv ln dg opts eo nsig nP uspec
+           anaUnitSpec lgraph libEnv ln dg opts eo rn nsig nP uspec 
+                                       -- TODO: is rn fine above?
```

Comment:
could you tell why it should / should not be fine?

Reply:
I am thinking we want 1. to follow some convention about the names of the nodes (with this argument we will get R_args0, ... R_argsn, R_res as node names for the unit specification being refined, which is weird) and 2. to make sure that we don't get the same node names twice (but now I think this is not possible, because the name R of the refinement is unique).

Reply:
Using rn here is bad, we get duplicate names. I'm looking into that, please do not merge until I fix it.